flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierAligner.java

```diff
@@ -318,6 +326,16 @@ public long getAlignmentDurationNanos() {
 		}
 	}
 
+	public void registerInputGate(CheckpointedInputGate inputGate) {
+		inputGates.add(checkNotNull(inputGate));
+	}
+
+	public void notifyCheckpointCompletedOrCanceled(long checkpointId) {
```

Comment:
nit: should be private method.

Reply:
We might need to adjust this method naming, because it is not indicating for the checkpoint really completed (the async checkpoint process). `notifyUnblockInputChannels`?